setpass/api.py

```diff
@@ -166,5 +168,46 @@ def add(user_id):
     return Response(response=user.token, status=200)
 
 
+@wsgi.app.route('/reset', methods=['GET'])
+def view_reset_form():
+    return render_template('reset_form.html')
+
+
+@wsgi.app.route('/reset', methods=['POST'])
+def reset_password():
+    name = request.form['name']
+    username = request.form['username']
+    pin = request.form['pin']
+
+    if not username or not pin:
+        return Response(response='Missing pin/username!', status=400)
+
+    _notify_helpdesk(name=name, username=username, pin=pin)
+
+    return Response(response='The request has been forwarded to the helpdesk.',
+                    status=200)
+
+
+def _notify_helpdesk(**kwargs):
+
+    with open('setpass/templates/helpdesk_ticket.txt', 'r') as f:
+        msg_body = f.read()
+    msg_body = msg_body.format(**kwargs)
+
+    sender = CONF['ticket_sender']
```

Comment:
Another optional suggestion, I think the canonical way of using oslo config stuff is `CONF.something` rather than `CONF['something']`